solidity/src/base/MathUtil.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+// This is licensed under the Cryptographic Open Software License 1.0
+pragma solidity ^0.8.28;
+
+/// A library for math utilities.
+library MathUtil {
+    /// Computes `max(1,ceil(log_2(value)))`,
```

Comment:
For docstrings, consider following the NatSpec format https://docs.soliditylang.org/en/latest/natspec-format.html